SocketRocket/SRWebSocket.m

```diff
@@ -1051,10 +1111,29 @@ - (void)_pumpWriting;
             [self _failWithError:error];
             return;
         }
-
+        
         _outputBufferOffset += bytesWritten;
+        
+        NSMutableArray<NSNumber *> *removeKeys = [NSMutableArray array];
+        [_sendCallbacks enumerateKeysAndObjectsUsingBlock:^(NSNumber * _Nonnull key, SRDataCallback * _Nonnull obj, BOOL * _Nonnull stop) {
+            if (NSMaxRange(obj.range) <= _outputBufferOffset) {
+                [removeKeys addObject:key];
+                if (obj.completeBlock) {
```

Comment:
This can go away once we can ensure that `_sendCallbacks` only contain objects with non-nil `completion`.